### DISCUSSION TITLE
abctl Local deployment: Can't login

### DISCUSSION THREAD

**Discussion by djodonnell** [airbytehq/airbyte#44021] [Deployment] [answered]
I've recently tried to setup Airbyte locally using Docker.

I performed the following steps, setting it up on MacOS:

(note: I already had Docker installed locally)

1. Install abctl command line tool 

```
brew tap airbytehq/tap
brew install abctl
```

2. Install airbyte locally:

`abctl local install`

This creates a docker image, starts a container and then opens a webpage prompting me to register a username and company.  I fill this in and then I am prompted to login

3. Retrieve the password

`abctl local credentials`

This returns the password along with a client-id and client-secret.  _Note that the password, client-id and client-secret included are all fabricated in this post._

```
{
  "password": "da095fbf-0212-4b3f-863e-901663c6b564",
  "client-id": "dfcaa21a-adc3-4af6-9d68-798b9805f798",
  "client-secret": "6055e608-4959-4ce6-9a47-4b575cf0c617"
}
```

4. Login to Airbyte

I attempt to use the username that I registered and the password copied from the previous step but it fails.

I verified the username is correct by issuing the following command:

`kubectl exec --kubeconfig ~/.airbyte/abctl/abctl,kubeconfig --namespace airbyte-abctl -it Airbyte-db-0 -- sql -U airbyte -d db-airbyte -t -A -c 'SELECT "email" FROM "user"'`

This command returns the correct username that I used.

Any clues or help would be appreciated as I'm really keen to explore Airbyte.

**Comment by marcosmarxm** [comment]
Hello 👋 , sorry you're having trouble installing Airbyte. I'll try to reproduce the same steps locally. 
I wasn't able to reproduce the problem locally as I was able to enter the application using the email.

@djodonnell can you please run the following command doing the necessary changes to your kubeconfig and database pod:
`kubectl exec --kubeconfig ~/.airbyte/abctl/abctl.kubeconfig --namespace airbyte-abctl -it airbyte-db-0 -- psql -U airbyte -d db-airbyte -t -A -c 'SELECT * FROM "organization"'`

The email used to access the application is the `organization` not the `user` table.

**Reply by gauravshah01** [reply]
Hello,
I had used the same email(returned by above command) for login still facing the same "Invalid username or password" issue as reported by OP.

Thanks

**Reply by marcosmarxm** [reply]
Can you contact me in Airbyte Slack @gauravshah01 ?

**Reply by djodonnell** [reply]
I tried selecting from the organisation table and the registered email address is the same one contained in users.

**Reply by my-mai** [reply]
I am getting the same issue as well. Everything is installed but for some reason the username (one registered) and password (generated) is not working. Invalid username or password error. 

Deployed on EC2 Linux

**Reply by marcosmarxm** [reply]
@djodonnell @my-mai can you contact me in Airbyte Slack so we can jump in a quick call?

**Reply by gauravshah01** [reply]
@marcosmarxm : Sure, please provide your slack handle or id where I can reach you

**Comment by bgroff** [comment]
Did you override the secret with a new value? If so can you run `kubectl rollout restart deployment/airbyte-abctl-server -n airbyte-abctl` and see if that resolves your issue?

**Reply by djodonnell** [reply]
I ran the command with the docker container running, and it returned the following error:

`E0815 10:30:54.140003   19614 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp [::1]:8080: connect: connection refused`

**Comment by marcosmarxm** [comment]
@djodonnell I was able to help @my-mai with the problem. In his situation he was accessing a Ec2 instance address that wasn't a HTTP. Because of that the Cookie configuration block the access. You said you're running locally right? But can confirm this is a local machine or are you trying a cloud provider? https://docs.airbyte.com/deploying-airbyte/integrations/authentication#cookie-security-settings

**Reply by djodonnell** [reply]
@marcosmarxm I am deploying it locally on MacOS using Docker Desktop

<img width="816" alt="image" src="https://github.com/user-attachments/assets/24d71507-bf52-4ca5-ba8e-2b51900aa309">

**Reply by marcosmarxm** [reply]
Can you try: `abctl local install --low-resource-mode --insecure-cookies`

**Comment by RayHughes** [comment]
Also deploying local on MacOS (14.5 (23F79)) with docker. Unable to login with password in credentials.

**Reply by marcosmarxm** [reply, marked as answer]
@RayHughes can you try `abctl local install --low-resource-mode --no-browser --insecure-cookies`

**Reply by RayHughes** [reply]
> @RayHughes can you try `abctl local install --low-resource-mode --no-browser --insecure-cookies`

This worked! Thank you!

**Reply by djodonnell** [reply]
Worked for me too - fantastic!

**Comment by techrick** [comment]
I had the same problems...
But with the optional flag `--insecure-cookies` it now works for me.

full example:
`abctl local install --low-resource-mode --no-browser --insecure-cookies`

trigger was the release notes of abctl [v0.12.0](https://github.com/airbytehq/abctl/releases/tag/v0.12.0):

What's Changed
* ...
* feat: add insecure cookies by @bgroff in https://github.com/airbytehq/abctl/pull/80
* ...

I think the flag `--insecure-cookies` flag should be enabled by default for local deployment, or not?

and by the way:
re-execute the `abctl local install` with the `--insecure-cookies` flag also works for me.

env:
* ubuntu 24.04
* docker v27.1.2
* abctl v0.12.0
* firefox v129.0.1 private mode
